### PR TITLE
Write test for `tmp_fname` to check if it accepts Path-like and str objects

### PR DESCRIPTION
**Pull request by algomaster99**
* [x] ❗ Have you followed the guidelines in the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) list?

* [x] 📖 Check this box if this PR **does not** require [documentation](https://dvc.org/doc) updates, or if it does **and** you have created a separate PR in [dvc.org](https://github.com/iterative/dvc.org) with such updates (or at least opened an issue about it in that repo). Please link below to your PR (or issue) in the [dvc.org](https://github.com/iterative/dvc.org) repo.

* [x] ❌ Have you checked DeepSource, CodeClimate, and other sanity checks below? We consider their findings recommendatory and don't expect everything to be addressed. Please review them carefully and fix those that actually improve code or fix bugs.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

Reference: #2137

### CODE REVIEW

**Comment by efiop** [comment]
Looks like we could also adjust https://github.com/iterative/dvc/blob/0.68.1/dvc/remote/hdfs.py#L128 to just pass `to_info`?

File: tests/unit/utils/test_utils.py [hunk 1]
```diff
@@ -122,3 +124,18 @@ def test_makedirs(repo_dir):
 
     makedirs(path_info)
     assert os.path.isdir(path_info.fspath)
+
+
+def test_tmp_fname():
+    file_path = os.path.join("path", "to", "file")
+    file_path_info = PathInfo(os.path.join("path-like", "to", "file"))
```
Comment (efiop) [inline tests/unit/utils/test_utils.py]:
```suggestion
    file_path_info = PathInfo(file_path)
```

**Comment by algomaster99** [comment]
@efiop `to_info` there isn't an object of `PathInfo` but `URLInfo`.

**Comment by efiop** [comment]
@algomaster99 Ah, good point.

**Pull request merged**